heron/healthmgr/src/java/com/twitter/heron/healthmgr/policy/AutoRestartBackpressureContainerPolicy.java

```diff
@@ -0,0 +1,105 @@
+import com.microsoft.dhalion.policy.HealthPolicyImpl;
+
+import com.twitter.heron.healthmgr.HealthPolicyConfig;
+import com.twitter.heron.healthmgr.common.HealthManagerEvents.ContainerRestart;
+import com.twitter.heron.healthmgr.detectors.BackPressureDetector;
+import com.twitter.heron.healthmgr.detectors.LargeWaitQueueDetector;
+import com.twitter.heron.healthmgr.detectors.ProcessingRateSkewDetector;
+import com.twitter.heron.healthmgr.detectors.WaitQueueDisparityDetector;
+import com.twitter.heron.healthmgr.diagnosers.DataSkewDiagnoser;
+import com.twitter.heron.healthmgr.diagnosers.SlowInstanceDiagnoser;
+import com.twitter.heron.healthmgr.diagnosers.UnderProvisioningDiagnoser;
+import com.twitter.heron.healthmgr.resolvers.RestartContainerResolver;
+
+import static com.twitter.heron.healthmgr.HealthPolicyConfigReader.PolicyConfigKey.HEALTH_POLICY_INTERVAL;
+import static com.twitter.heron.healthmgr.diagnosers.BaseDiagnoser.DiagnosisName.DIAGNOSIS_DATA_SKEW;
+import static com.twitter.heron.healthmgr.diagnosers.BaseDiagnoser.DiagnosisName.DIAGNOSIS_SLOW_INSTANCE;
+import static com.twitter.heron.healthmgr.diagnosers.BaseDiagnoser.DiagnosisName.DIAGNOSIS_UNDER_PROVISIONING;
+
+public class AutoRestartBackpressureContainerPolicy extends HealthPolicyImpl
+    implements EventHandler<ContainerRestart> {
+
+  private static final String CONF_WAIT_INTERVAL_MILLIS =
+      "AutoRestartBackpressureContainerPolicy.conf_post_action_wait_interval_min";
+  private static final Logger LOG
+      = Logger.getLogger(AutoRestartBackpressureContainerPolicy.class.getName());
+  
+  private HealthPolicyConfig policyConfig;
+  private RestartContainerResolver restartContainerResolver;
+
+  @Inject
+  AutoRestartBackpressureContainerPolicy(HealthPolicyConfig policyConfig,
+                                  EventManager eventManager,
+                                  BackPressureDetector backPressureDetector,
+                                  LargeWaitQueueDetector largeWaitQueueDetector,
+                                  ProcessingRateSkewDetector dataSkewDetector,
+                                  WaitQueueDisparityDetector waitQueueDisparityDetector,
+                                  UnderProvisioningDiagnoser underProvisioningDiagnoser,
+                                  DataSkewDiagnoser dataSkewDiagnoser,
+                                  SlowInstanceDiagnoser slowInstanceDiagnoser,
+                                  RestartContainerResolver restartContainerResolver) {
+    this.policyConfig = policyConfig;
+    this.restartContainerResolver = restartContainerResolver;
+
+    registerDetectors(backPressureDetector, largeWaitQueueDetector,
+        waitQueueDisparityDetector, dataSkewDetector);
+    registerDiagnosers(underProvisioningDiagnoser, dataSkewDiagnoser, slowInstanceDiagnoser);
+
+    setPolicyExecutionInterval(TimeUnit.MILLISECONDS,
+        (int) policyConfig.getConfig(HEALTH_POLICY_INTERVAL.key(), 60000));
```

Comment:
Can we use a more type-safe method rather than a cast `(int) policyConfig.getConfig....`?